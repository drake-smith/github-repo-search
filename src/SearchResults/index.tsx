import React from 'react';
import styled from 'styled-components';
import { media } from '../shared/media';

const { tablet } = media;

interface SearchResultsProps {
  results: Array<any>;
}

const SearchResultsContainer = styled.div`
  width: 100%;
`;

const SearchResults = ({ results }: SearchResultsProps) => {
  return (
    <SearchResultsContainer>
      {results.map((result, i) => {
        return (
          <div key={i}>
            <h2>suh</h2>
            <p>dude</p>
          </div>
        );
      })}
    </SearchResultsContainer>
  );
};

export default SearchResults;
