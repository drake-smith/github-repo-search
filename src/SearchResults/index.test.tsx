import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import SearchResults from '.';

describe('<SearchResults />', () => {
  const handleSelectSearchResult = jest.fn();
  const mockObj = [
    {
      id: 1,
      name: 'React',
      owner: {
        login: 'facebook',
      },
      description: 'This is a library.',
      stargazers_count: 100,
      language: 'JavaScript',
    },
    {
      id: 2,
      name: 'Angular',
      owner: {
        login: 'google',
      },
      description: 'This is a library.',
      stargazers_count: 100,
      language: 'JavaScript',
    },
  ];
  test('It renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <SearchResults
            results={mockObj}
            handleSelectSearchResult={handleSelectSearchResult}
          />
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
