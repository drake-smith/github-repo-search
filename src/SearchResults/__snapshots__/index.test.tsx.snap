// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchResults /> It renders correctly 1`] = `
<ul
  className="sc-jRQBWg fMshSK"
>
  <li
    className="sc-bdvvtL bBGwRo"
  >
    <div>
      <a
        className="sc-gsDKAQ peeim"
        href="/1"
        onClick={[Function]}
      >
        <h2
          className="sc-dkPtRN fCvFVC"
        >
          React
        </h2>
      </a>
      <p
        className="sc-hKwDye jzfCYy"
      >
        This is a library.
      </p>
    </div>
    <div>
      <div
        className="sc-eCImPb iMXmGk"
      >
        <p>
          by 
          <strong>
            facebook
          </strong>
        </p>
        <p>
          100
           ⭐ 
        </p>
        <p>
          JavaScript
        </p>
      </div>
    </div>
  </li>
  <li
    className="sc-bdvvtL bBGwRo"
  >
    <div>
      <a
        className="sc-gsDKAQ peeim"
        href="/2"
        onClick={[Function]}
      >
        <h2
          className="sc-dkPtRN fCvFVC"
        >
          Angular
        </h2>
      </a>
      <p
        className="sc-hKwDye jzfCYy"
      >
        This is a library.
      </p>
    </div>
    <div>
      <div
        className="sc-eCImPb iMXmGk"
      >
        <p>
          by 
          <strong>
            google
          </strong>
        </p>
        <p>
          100
           ⭐ 
        </p>
        <p>
          JavaScript
        </p>
      </div>
    </div>
  </li>
</ul>
`;
